fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,497 @@
+        patches_field (None): a sample field defining the image patches in each
+            sample that have been/will be embedded. Must be of type
+            :class:`fiftyone.core.labels.Detection`,
+            :class:`fiftyone.core.labels.Detections`,
+            :class:`fiftyone.core.labels.Polyline`, or
+            :class:`fiftyone.core.labels.Polylines`
+        embeddings_field (None): field for embeddings to feed the index. This argument's
+            behavior depends on whether a ``model`` is provided, as described
+            below.
+
+            If no ``model`` is provided, this argument specifies the field of pre-computed
+            embeddings to use:
+
+            If a ``model`` is provided, this argument specifies where to store
+            the model's embeddings:
+        brain_key (None): a brain key under which to store the results of this
+            method
+        model (None): a :class:`fiftyone.core.models.Model` or the name of a
+            model from the
+            `FiftyOne Model Zoo <https://docs.voxel51.com/user_guide/model_zoo/index.html>`_
+            to use, or that was already used, to generate embeddings. The model
+            must expose embeddings (``model.has_embeddings = True``)
+        model_kwargs (None): a dictionary of optional keyword arguments to pass
+            to the model's ``Config`` when a model name is provided
+        **kwargs: keyword arguments for the
+            :class:`fiftyone.brain.SklearnSimilarityIndex` class
+
+    Returns:
+        a :class:`fiftyone.brain.internal.core.leaky_splits.LeakySplitsSKLIndex`, a :class:`fiftyone.core.view.DatasetView`
+    """
+
+    fov.validate_collection(samples)
+
+    embeddings_exist = False
+    if embeddings_field is not None and model is None:
+        embeddings_field, embeddings_exist = fbu.parse_embeddings_field(
+            samples,
+            embeddings_field,
+            patches_field=patches_field,
+        )
+
+    config = LeakySplitsSKLConfig(
+        split_views=split_views,
+        split_field=split_field,
+        split_tags=split_tags,
+        embeddings_field=embeddings_field,
+        model=model,
+        model_kwargs=model_kwargs,
+        patches_field=patches_field,
+        metric=metric,
+        **kwargs,
+    )
+    brain_method = LeakySplitsSKL(config)
+    brain_method.ensure_requirements()
+
+    if brain_key is not None:
+        # Don't allow overwriting an existing run with same key, since we
+        # need the existing run in order to perform workflows like
+        # automatically cleaning up the backend's index
+        brain_method.register_run(samples, brain_key, overwrite=False)
+
+    results = brain_method.initialize(samples, brain_key)
+
+    results.set_threshold(threshold)
+    leaks = results.leaks
+
+    brain_method.save_run_results(samples, brain_key, results)
+
+    return results, leaks
+
+
+### GENERAL
+
+
+class LeakySplitsConfigInterface(object):
```

Comment:
Mixins are kind of like that. The reason I went for this design choice is so that I only have to avoid code duplication for the overlap between the similarity and hash backends. I couldn't really find a clean way to fit them both into one class. If you have a suggestion I'd be happy to refactor.

Reply:
Will no longer be an issue after [refactor](https://github.com/voxel51/fiftyone-brain/pull/203#discussion_r1849036659). Hash will be removed so no longer any need to consolidate common functionality in one spot.